### ISSUE TITLE
DeferredRegister vs RegisterEvent recommendation

### ISSUE THREAD

**Issue by naxhh** [MinecraftForge/Documentation#492]
https://github.com/MinecraftForge/Documentation/blob/1.19.x/docs/concepts/registries.md#methods-for-registering

states

> DeferredRegister is the recommended way to register objects

to later say

> RegisterEvent [...] It is recommended to use this method to avoid unnecessary object creation

For someone new that is trying to learn the direct question is: "which one is the recommended one then?"

DeferredRegister because is easier to setup and says "recommended"

or RegisterEvent because it will not cause as many as object registries and that is really important in this case?

Maybe rewording one of them to make them more clear would help

**Comment by ChampionAsh5357** [comment]
`DeferredRegister` is the recommended method. The reason to use `RegisterEvent` is if you don't need to refer to the object being created within your source or you are creating an optional entry, typically for mod interoperability.
